unblob/file_utils.py

```diff
@@ -53,6 +59,8 @@ def from_bytes(cls, content: bytes):
 
     @classmethod
     def from_path(cls, path: Path, access=mmap.ACCESS_READ):
+        if path.stat().st_size == 0:
+            raise InvalidInputFormat("Can't create File from empty file.")
         mode = "r+b" if access == mmap.ACCESS_WRITE else "rb"
         with path.open(mode) as base_file:
             m = cls(base_file.fileno(), 0, access=access)
```

Comment:
I think we can just catch the `ValueError("cannot mmap an empty file")` from mmap here:

```suggestion
        mode = "r+b" if access == mmap.ACCESS_WRITE else "rb"
        with path.open(mode) as base_file:
            try:
                m = cls(base_file.fileno(), 0, access=access)
            except ValueError as exc:
                raise InvalidInputFormat from exc
```